homotopy-graphics/src/style.rs

```diff
@@ -0,0 +1,22 @@
+use homotopy_core::Generator;
+
+pub trait GeneratorStyle {
+    fn label(&self) -> Option<String>;
+    fn shape(&self) -> Option<VertexShape>;
+    // TODO(thud): migrate color-related code to here for consistency
+}
+
+pub trait GeneratorStyles<T: GeneratorStyle> {
```

Comment:
This is a great way to avoid the dependency problem, well done!

Have you considered moving the type `T` inside the trait to avoid having so many generics everywhere, i.e.

```rust
pub trait GeneratorStyles {
    type T: GeneratorStyle;

    fn generator_style(&self, g: Generator) -> Option<&Self::T>;
}
```

Also (and this is more of a nitpick) could we find a different name? How about `Signature` or `SignatureStylesheet`?